### ISSUE TITLE
Fail loudly when building the DAO

### ISSUE THREAD

**Issue by Crevil** [jonbodner/proteus#16]
It would be great if we had an option to build a DAO and be certain that all fields are mapped as expected. Right now a warning is written to the logs but the building continues.

I'm thinking something like a `proteus.MustBuild()` function that returns an error on any encountered error. What du you think about this? Would you be interested in this feature if we provide a PR :)

**Comment by jonbodner** [comment]
Do you want a panic on failure? Would you want an error that contains all of the errors that were found? (I can whip up a version of multierror that has a better license than the Hashicorp one) And if you already have one, that's great too 👍

**Comment by Crevil** [comment]
I prefer `error`. A simple `error` type exposing all errors found that can be inspected would be great. Possibly it could be an exported type or function to receive the errors.

On the top of my head, I'm thinking something like
```go
err := proteus.Build(dao, proteus.Postgres)
if err != nil {
  if pErr, ok := err.(proteus.Error); ok {
    for _, p := range pErr.Errors() {
      ...
    }
  }
}
```

**Comment by jonbodner** [comment]
Sounds great. What do you think of this for the multierr: https://github.com/jonbodner/multierr

**Comment by Crevil** [comment]
Looks good and simple. Could easily work. It's maybe simple enough to just include directly with `proteus` if you don't mind?

**Comment by jonbodner** [comment]
I still need to add more testing, but here's the first implementation of this feature: https://github.com/jonbodner/proteus/tree/add_ShouldBuild

You can access it with git hash fb18f2c37b7a69d58334436fb0d146eb4e7d77cd

I also have code on this branch to allow anonymous access to structs and maps, so you can now say $1.Name instead of specifying a name using prop struct tags.

**Comment by Crevil** [comment]
This looks great. 👍

**Comment by jonbodner** [comment]
I'm going to mark this as closed, as version 0.5.2 fixes this issue. If it doesn't, please reopen. Note that the ShouldBuild API is going to change slightly in the next release (either 0.5.3 or 0.6.0); a context.Context first parameter is going to be added. Apologies for not having this in the earlier version.

**Comment by Crevil** [comment]
I've upgraded to 0.5.2 and it's great! No worries about the changes. That's why it stil 0.x version ;)
The comming changes looks good as well.
